ql/src/java/org/apache/hadoop/hive/ql/optimizer/ConvertJoinMapJoin.java

```diff
@@ -755,6 +755,14 @@ private boolean checkConvertJoinSMBJoin(JoinOperator joinOp, OptimizeTezProcCont
         LOG.debug("External table {} found in join and also could not provide statistics - disabling SMB join.", sb);
         return false;
       }
+      for (Operator<?> grandParent : parentOp.getParentOperators()) {
```

Comment:
Similar to the other other method,  pls add a non-null check:
 if (parentOp.getParentOperators() != null) {
    for ( ...)

Reply:
Added, thanks.